### ISSUE TITLE
range-builtin-not-iterating false positive when used with itertools

### ISSUE THREAD

**Issue by Bernhard10** [PyCQA/pylint#2731]
### Steps to reproduce
Create test.py:
```
from itertools import product
for i,j in product(range(4), repeat=2):
    pass
```
run pylint test.py --enable range-builtin-not-iterating

### Current behavior
Gives a `range-builtin-not-iterating` warning.

### Expected behavior
 `range-builtin-not-iterating` is not triggered, because functions in itertools accept iterators.

### pylint --version output

__main__.py 1.9.4, 
astroid 1.6.5
Python 2.7.15 (default, Oct 15 2018, 15:24:06) 
[GCC 8.1.1 20180712 (Red Hat 8.1.1-5)]
